packages/imperative/src/config/cmd/convert-profiles/convert-profiles.handler.ts

```diff
@@ -0,0 +1,195 @@
+        ConfigSchema.updateSchema();
+        await teamConfig.save(false);
+
+        const oldProfilesDir = `${profilesRootDir.replace(/[\\/]$/, "")}-old`;
+        try {
+            fs.renameSync(profilesRootDir, oldProfilesDir);
+        } catch (error) {
+            params.response.console.error(`Failed to rename profiles directory to ${oldProfilesDir}:\n    ${error}`);
+        }
+
+        const cliBin = ImperativeConfig.instance.rootCommandName;
+        // TODO Implement config edit command
+        params.response.console.log(`Your new profiles have been saved to ${teamConfig.layerActive().path}.\n` +
+            `Run "${cliBin} config edit --global-config" to open this file in your default editor.\n\n` +
+            `Your old profiles have been moved to ${oldProfilesDir}.\n` +
+            `Run "${cliBin} config convert-profiles --delete" if you want to completely remove them.`);
+    }
+
+    /**
+     * Retrieve list of obsolete CLI plug-ins that should be uninstalled.
+     * @returns List of plugins to uninstall containing the following properties:
+     *  - `name` - Name of plugin to uninstall
+     *  - `preuninstall` - Optional callback to be run before uninstall
+     *  - `postuninstall` - Optional callback to be run after uninstall
+     */
+    private getObsoletePlugins(): IPluginToUninstall[] {
+        const obsoletePlugins: IPluginToUninstall[] = [];
+
+        if (ImperativeConfig.instance.hostPackageName === this.ZOWE_CLI_PACKAGE_NAME) {
+            let credMgrSetting = AppSettings.instance.get("overrides", "CredentialManager");
+            if (credMgrSetting === ImperativeConfig.instance.hostPackageName) {
+                credMgrSetting = undefined;
+            }
+            obsoletePlugins.push({
+                name: typeof credMgrSetting === "string" ? credMgrSetting : this.ZOWE_CLI_SECURE_PLUGIN_NAME,
+                preUninstall: credMgrSetting ? this.disableCredentialManager : undefined
+            });
+        }
+
+        return obsoletePlugins;
+    }
+
+    /**
+     * Get the number of old profiles present in the CLI home dir.
+     * @param profilesRootDir Root profiles directory
+     * @returns Number of old profiles found
+     */
+    private getOldProfileCount(profilesRootDir: string): number {
+        const profileTypes = ProfileIO.getAllProfileDirectories(profilesRootDir);
+        let oldProfileCount = 0;
+        for (const profileType of profileTypes) {
+            const profileTypeDir = path.join(profilesRootDir, profileType);
+            const profileNames = ProfileIO.getAllProfileNames(profileTypeDir, ".yaml", `${profileType}_meta`);
+            oldProfileCount += profileNames.length;
+        }
+        return oldProfileCount;
+    }
+
+    /**
+     * Disable the CredentialManager override in app settings. This is called
+     * before uninstalling `ZOWE_CLI_SECURE_PLUGIN_NAME`.
+     */
+    private disableCredentialManager() {
```

Comment:
Should this be something moved to an API instead of being a private function ?

Reply:
Will add JSDoc explaining that this method only works when CredentialManager has not been frozen ❄️